manifests/source.pp

```diff
@@ -1,65 +1,46 @@
 # source.pp
 # add an apt source
 define apt::source(
-  $location          = undef,
-  $comment           = $name,
-  $ensure            = present,
-  $release           = undef,
-  $repos             = 'main',
-  $include           = {},
-  $key               = undef,
-  $pin               = undef,
-  $architecture      = undef,
-  $allow_unsigned    = false,
-  $include_src       = undef,
-  $include_deb       = undef,
-  $required_packages = undef,
-  $key_server        = undef,
-  $key_content       = undef,
-  $key_source        = undef,
-  $trusted_source    = undef,
-  $notify_update     = true,
+  Optional[Variant[String, Stdlib::Compat::String]] $location                         = undef,
+  Variant[String, Stdlib::Compat::String] $comment                                    = $name,
+  Variant[String, Stdlib::Compat::String] $ensure                                     = present,
+  Optional[Variant[String, Stdlib::Compat::String]] $release                          = undef,
+  Variant[String, Stdlib::Compat::String] $repos                                      = 'main',
+  Optional[Variant[Hash, Stdlib::Compat::Hash]] $include                              = {},
+  Optional[Variant[String, Stdlib::Compat::String, Hash, Stdlib::Compat::Hash]] $key  = undef,
+  $pin                                                                                = undef,
+  Optional[Variant[String, Stdlib::Compat::String]] $architecture                     = undef,
+  Boolean $allow_unsigned                                                             = false,
+  Boolean $notify_update                                                              = true,
+  Optional[Variant[String, Stdlib::Compat::String]] $key_server                       = undef,
+  Optional[Variant[String, Stdlib::Compat::String]] $key_content                      = undef,
+  Optional[Variant[String, Stdlib::Compat::String]] $key_source                       = undef,
```

Comment:
Is the next major release soon? Are more breaking changes than this one planned? (required_packages looks nasty, and trusted_source is a simple fix/change.)

The deprecation warnings have been around since apt 2.0.0 was released two years ago, so it seems like anyone currently using 1.8 or higher will have migrated up unless they were tied to 1.x, so as far as a breaking change goes this would not be breaking something that people depend on and haven't had ample time to move on, and Morgan provided a migration path back in the 1.8/2.0 to make this easier to do.

Looking through our modules, I see https://github.com/puppetlabs/puppetlabs-rabbitmq/blob/master/manifests/repo/apt.pp is using it. Lovely.

(I was originally asking because you removed the deprecation warnings below. If we're not deprecating them, then we won't be removing them, and that could just be an easy answer... I wonder why they were deprecated in the first place?)